### PR TITLE
[RFR] Make Postcommit tasks marked as celery run as a chain

### PR DESCRIPTION
**Pull request by cwisecarver**
## Purpose

Currently, postcommit tasks could run in any order, asynchronously. This changes them to run as a celery chain: synchronously and in order.

## Changes

There is a separate queue for celery specific postcommit tasks. This queue is run as a celery chain after the request's transaction is committed successfully.

## Side effects

Tasks decorated with run_postcommit could be delayed to the point where a user would notice.

### CODE REVIEW

File: framework/postcommit_tasks/handlers.py [hunk 1]
```diff
@@ -70,9 +95,6 @@ def wrapper(func):
             return func
         @functools.wraps(func)
         def wrapped(*args, **kwargs):
-            if celery is True and isinstance(func, PromiseProxy):
-                func.delay(*args, **kwargs)
-            else:
-                enqueue_postcommit_task(func, args, kwargs, once_per_request=once_per_request)
+            enqueue_postcommit_task(func, args, kwargs, celery=celery, once_per_request=once_per_request)
```
Comment (brianjgeiger) [inline framework/postcommit_tasks/handlers.py]: 👍

File: framework/postcommit_tasks/handlers.py [hunk 2]
```diff
@@ -35,12 +48,20 @@ def postcommit_after_request(response, base_status_error_code=500):
             for func in postcommit_queue().values():
                 pool.spawn(func)
             pool.join(timeout=5.0, raise_error=True)  # 5 second timeout and reraise exceptions
+
+        if postcommit_celery_queue():
+            if settings.USE_CELERY:
+                postcommit_celery_task_wrapper.delay(postcommit_celery_queue())
```
Comment (cwisecarver) [inline framework/postcommit_tasks/handlers.py]: This is the top-secret bad way to get around the maximum recursion bug. Make a task that calls all the other tasks as a chain and call that, passing in the signatures of the other tasks. I read about it on the internet.

Comment (brianjgeiger) [inline framework/postcommit_tasks/handlers.py]: Is it worth linking to the source in a code comment for people who later want to know about this?

Comment (cwisecarver) [inline framework/postcommit_tasks/handlers.py]: dun

**Pull request closed without merging**